src/CMakeLists.txt

```diff
@@ -1,16 +1,16 @@
 include_directories(.)
 include_directories(../include)
 
+add_subdirectory(autotuner)
+add_subdirectory(core)
```

Comment:
core depends on version, lang, proto so we might want to move this line after add_subdirectory(version). We should prioritize the order of build execution rather than naming :)

Reply:
So I'd claim that the ordering of dependencies is a lexicographic order that cmake solves so order of subdirs shouldn't matter.
The issue I had that triggered this is interleaving of WITH_CUDA and non WITH_CUDA behavior which made me hoist ```autotuner``` before ```aten```; I'm not sure I'd want to change that.

In the light of the above do you still feel autotuner should go back at then end?

I'm mostly worried about interleaving of these things when we also add opencl and more behaviors. 
It will then look like spaghetti interleaving but I can see that this is premature optimization at this point :)

Reply:
oh I meant core actually :) autotuner is okay

Reply:
Ok so then I'll upate with this lexicographic rule:
1. no spaghetti options
2. order of dependencies rather than dictionary order

Thanks!